src/Microsoft.DocAsCode.App/RunServe.cs

```diff
@@ -43,11 +47,104 @@ public static void Exec(string folder, string host, int? port)
             Console.WriteLine($"Serving \"{folder}\" on {url}. Press Ctrl+C to shut down.");
             using var app = builder.Build();
             app.UseFileServer(fileServerOptions);
-            app.Run();
+
+            if (openBrowser)
+            {
+                string relativePath = openBrowserRelativePath;
+                var launchUrl = string.IsNullOrEmpty(relativePath)
+                    ? url
+                    : Path.Combine(url, ResolveOutputHtmlRelativePath(folder, relativePath));
+
+                // Start web server.
+                app.Start();
+
+                // Launch browser process.
+
+                Console.WriteLine($"Launching browser with url: {url}.");
+                using var process = LaunchBrowser(launchUrl);
+
+                // Wait until server exited.
+                app.WaitForShutdown();
+
+                // process object is disposed. (Note:Launched process remain running.after dispose)
+            }
+            else
+            {
+                app.Run();
+            }
         }
         catch (System.Reflection.TargetInvocationException)
         {
             Logger.LogError($"Error serving \"{folder}\" on {url}, check if the port is already being in use.");
         }
     }
+
+    /// <summary>
+    /// Resolve output HTML file path by `manifest.json` file.
+    /// </summary>
+    private static string ResolveOutputHtmlRelativePath(string folder, string relativePath)
+    {
+        var manifestPath = Path.GetFullPath(Path.Combine(folder, "manifest.json"));
+        if (!File.Exists(manifestPath))
+            return string.Empty;
+
+        try
+        {
+            Manifest manifest = JsonUtility.Deserialize<Manifest>(manifestPath);
+
+            // Try to find output html file (html->html)
+            OutputFileInfo outputFileInfo = manifest.FindOutputFileInfo(relativePath);
+            if (outputFileInfo != null)
+                return outputFileInfo.RelativePath;
+
+            // Try to resolve output HTML file. (md->html)
+            relativePath = relativePath.Replace('\\', '/'); // Normalize path.
+            var manifestFile = manifest.Files
+                                       .Where(x => FilePathComparer.OSPlatformSensitiveRelativePathComparer.Equals(x.SourceRelativePath, relativePath))
+                                       .FirstOrDefault(x => x.OutputFiles.TryGetValue(".html", out outputFileInfo));
+
+            if (outputFileInfo != null)
+                return outputFileInfo.RelativePath;
+        }
+        catch (Exception)
```

Comment:
consider remove this `try/catch/throw`, there behavior is almost the same with no exception handling as all exceptions are thrown immediately.

Reply:
Changed behavior to log errors. then fallback to return baseUrl.